### PR TITLE
OPENEUROPA-2774: [Event] Write behat test for time conditional logic

### PR DESCRIPTION
**Pull request by ademarco**

### CODE REVIEW

File: modules/oe_theme_content_event/templates/oe-theme-content-event-contacts.html.twig [hunk 1]
```diff
@@ -7,19 +7,29 @@
...
-        {{ general }}
-      </div>
-    {% endif %}
-    {% if press is not empty %}
-      <div class="ecl-col-12 ecl-col-md-6{{ general is not empty ? ' ecl-u-mt-l ecl-u-mt-md-none'}}">
-        <h3 class="ecl-u-type-heading-3 ecl-u-type-color-black ecl-u-mt-none ecl-u-mb-m ecl-u-mb-md-l">{{ 'Press contact'|t }}</h3>
-        {{ press }}
-      </div>
-    {% endif %}
+  <div id="event-contacts">
```
Comment (brummbar) [inline modules/oe_theme_content_event/templates/oe-theme-content-event-contacts.html.twig:10]: Are these IDs needed only for the tests?

File: templates/content/field-group-html-element--node--oe-event--group-middle.html.twig [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{#
+/**
+ * @file
+ * Default theme implementation for a fieldgroup html element.
```
Comment (brummbar) [inline templates/content/field-group-html-element--node--oe-event--group-middle.html.twig]: Let's say it's an override.

File: tests/Behat/DrupalContext.php [hunk 3]
```diff
@@ -77,4 +82,42 @@ public function selectOption(string $select, string $option, string $region): vo
...
+   *
+   * @param string $title
+   *   Content title.
+   *
+   * @Given I am visiting the :title content
+   * @Given I visit the :title content
+   */
+  public function iAmViewingTheContent($title): void {
+    $nid = $this->loadEntityByLabel('node', $title)->id();
+    $this->visitPath("node/$nid");
```
Comment (brummbar) [inline tests/Behat/DrupalContext.php]: $node->toUrl()

File: tests/Behat/EventContext.php [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    $this->assertRegistrationButtonExists();
+    if ($this->getRegistrationButton()->getTagName() === 'a') {
+      throw new \Exception('The registration button was not supposed to be active.');
+    }
+  }
+
+  /**
+   * Assert whether the registration button exists.
+   *
+   * @Then the registration button should be there
```
Comment (brummbar) [inline tests/Behat/EventContext.php]:
```suggestion
   * @Then the registration button should be visible/present
```

File: tests/features/event.feature [hunk 5]
```diff
@@ -1,147 +1,273 @@
...
+    And I should see the text "Report summary"
+    And I should see the text "Report text"
+    But I should not see the heading "Description"
+    And I should not see the text "Event description summary"
+    And I should not see the text "Event description"
+    And I should not see the registration block
+
+  @preserve_anonymous_permissions
+  Scenario: As an anonymous user, when I visit an event I can see the information in the correct layout
+    And I am on the homepage
```
Comment (brummbar) [inline tests/features/event.feature]:
```suggestion
    Given I am on the homepage
```

File: tests/features/event.feature [hunk 6]
```diff
@@ -1,147 +1,273 @@
...
+    And the following Event Content entity:
+      | Title                   | Event demo page                                               |
+      | Type                    | exhibitions                                                   |
+      | Introduction            | Event introduction text                                       |
+      | Description summary     | Description summary text                                      |
+      | Description             | Event description                                             |
+      | Start date              | 2019-02-21T02:21:00                                           |
+      | End date                | 2019-02-21T14:21:00                                           |
+      | Status                  | as_planned                                                    |
+      | Languages               | http://publications.europa.eu/resource/authority/language/0D0 |
```
Comment (brummbar) [inline tests/features/event.feature]: Why there's the ID here?

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/RegistrationButtonExtraField.php [hunk 7]
```diff
@@ -87,45 +92,53 @@ public function viewElements(ContentEntityInterface $entity) {
...
+        '@start_date' => $this->dateFormatter->format($datetime_start->getTimestamp(), 'oe_event_date_hour'),
+        '@end_date' => $this->dateFormatter->format($datetime_end->getTimestamp(), 'oe_event_date_hour'),
       ]);
+      $build['#enabled'] = FALSE;
 
       // We invalidate this message every day at midnight.
-      $this->applyMidnightMaxAge($build, $event);
+      $this->applyMidnightTag($build, $datetime_start);
+
+      // We invalidate this message when the registration period starts.
```
Comment (brummbar) [inline modules/oe_theme_content_event/src/Plugin/ExtraField/Display/RegistrationButtonExtraField.php:113]: Shouldn't we say that we invalidate every hour?

**Review by brummbar — CHANGES_REQUESTED** [review]

File: modules/oe_theme_content_event/templates/oe-theme-content-event-contacts.html.twig (hunk 1 above)
Comment (ademarco) [reply modules/oe_theme_content_event/templates/oe-theme-content-event-contacts.html.twig:10]: Yep, that's the case.

File: templates/content/field-group-html-element--node--oe-event--group-middle.html.twig (hunk 2 above)
Comment (ademarco) [reply templates/content/field-group-html-element--node--oe-event--group-middle.html.twig]: Done.

File: modules/oe_theme_content_event/templates/oe-theme-content-event-contacts.html.twig (hunk 1 above)
Comment (ademarco) [reply modules/oe_theme_content_event/templates/oe-theme-content-event-contacts.html.twig:10]: So we agreed with @brummbar to leave it like this for now.

File: tests/Behat/EventContext.php (hunk 4 above)
Comment (ademarco) [reply tests/Behat/EventContext.php]: Done.

File: tests/features/event.feature (hunk 5 above)
Comment (ademarco) [reply tests/features/event.feature]: Crap that's a leftover from using `@javascript` for development, removing.

File: tests/Behat/DrupalContext.php (hunk 3 above)
Comment (ademarco) [reply tests/Behat/DrupalContext.php]: Done.

File: tests/features/event.feature (hunk 6 above)
Comment (ademarco) [reply tests/features/event.feature]: Using language label now.

**Review by nagyad — DISMISSED** [review]

**Review by brummbar — DISMISSED** [review]

**Review by brummbar — APPROVED** [review]

**Review by nagyad — APPROVED** [review]

**Pull request merged**